db: TestIngestLoadRand is flaky

See [failure on a master build](https://github.com/cockroachdb/pebble/actions/runs/6437872534/job/17483653532).


```
--- FAIL: TestIngestLoadRand (0.00s)
    ingest_test.go:223: [5].LargestPointKey.UserKey[0]: uint8(0x76) (previously visited) != uint8(0x76)
        [5].LargestPointKey.UserKey[1]: uint8(0x85) (previously visited) != uint8(0x85)
        [5].LargestPointKey.UserKey[2]: uint8(0xa) (previously visited) != uint8(0xa)
        [5].LargestPointKey.UserKey[3]: uint8(0xb1) (previously visited) != uint8(0xb1)
        [5].Smallest.UserKey[0]: uint8(0x76) (previously visited) != uint8(0x76)
        [5].Smallest.UserKey[1]: uint8(0x85) (previously visited) != uint8(0x85)
        [5].Smallest.UserKey[2]: uint8(0xa) (previously visited) != uint8(0xa)
        [5].Smallest.UserKey[3]: uint8(0xb1) (previously visited) != uint8(0xb1)
        [5].Largest.UserKey[0]: uint8(0x76) (previously visited) != uint8(0x76)
        [5].Largest.UserKey[1]: uint8(0x85) (previously visited) != uint8(0x85)
        [5].Largest.UserKey[2]: uint8(0xa) (previously visited) != uint8(0xa)
        [5].Largest.UserKey[3]: uint8(0xb1) (previously visited) != uint8(0xb1)
```

--- a/ingest_test.go
+++ b/ingest_test.go
@@ -219,9 +219,8 @@ func TestIngestLoadRand(t *testing.T) {
 	for _, m := range lr.localMeta {
 		m.CreationTime = 0
 	}
-	if diff := pretty.Diff(expected, lr.localMeta); diff != nil {
-		t.Fatalf("%s", strings.Join(diff, "\n"))
-	}
+	t.Log(strings.Join(pretty.Diff(expected, lr.localMeta), "\n"))
+	require.Equal(t, expected, lr.localMeta)
 }
 
 func TestIngestLoadInvalid(t *testing.T) {

